### PR TITLE
Add Translation Memory tab to the team page

### LINKED ISSUE #3031: Add ability to edit Translation Memory entries
Sometimes TM entries contain bugs and only Admins can fix them. 

Managers and Translators, who can create entries, should at least be allowed to delete them. Ideally they should also be able to make edits.

See also #2031.

### PR DESCRIPTION
**Pull request by mathjazz**
Fix #3031.

This is the main step towards Translation Memory Management ([spec](https://github.com/mozilla/pontoon/blob/main/specs/0121-translation-memory-management.md)). The missing part is .TMX file upload, tracked under #2031. The PR is deployed to [stage](https://mozilla-pontoon-staging.herokuapp.com/sl/translation-memory/) for testing.

This is what it does:
* List all TM entries for the locale
* Markup empty strings to make them accessible in translate view
* Implement infinite scroll with skeleton loader
* Add ability to search TM entries with match highlighting
* Add ability to Delete TM entries
* Add ability to Edit TM entries

We should file issues for the followups:
- [ ] Reflect search and pagination in URL parameters
- [ ] Highlight placeables
- [x] Refactor ActionLog to use [Generic relationships](https://docs.djangoproject.com/en/5.1/ref/contrib/contenttypes/#generic-relations)

### CODE REVIEW

**Review by eemeli — APPROVED** [review]
See inline for some nitpicks.

File: pontoon/actionlog/models.py [hunk 1]
```diff
@@ -58,31 +70,59 @@ def validate_action_type_choice(self):
...
+            if self.translation or not self.entity or not self.locale:
+                raise ValidationError(
+                    f'For action type "{self.action_type}", only `entity` and `locale` are accepted'
+                )
+
+        elif self.action_type == self.ActionType.COMMENT_ADDED:
+            if not (
+                (self.translation and not self.locale and not self.entity)
+                or (not self.translation and self.locale and self.entity)
+            ):
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/actionlog/models.py:86]: Filed https://github.com/mozilla/pontoon/issues/3443 to refactor ActionLog.

File: pontoon/actionlog/models.py [hunk 2]
```diff
@@ -58,31 +70,59 @@ def validate_action_type_choice(self):
             )
 
     def validate_foreign_keys_per_action(self):
-        if self.action_type == self.ActionType.TRANSLATION_DELETED and (
-            self.translation or not self.entity or not self.locale
+        if self.action_type in (
+            self.ActionType.TRANSLATION_DELETED,
+            self.ActionType.TM_ENTRY_DELETED,
         ):
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/actionlog/models.py:76]: Filed https://github.com/mozilla/pontoon/issues/3443 to refactor ActionLog.

**Comment by flodolo** [comment]
Where is documentation for this feature tracked?
https://github.com/mozilla/pontoon/pull/3437

**Comment by mathjazz** [comment]
We don't have an issue on file, but the PR will be ready shortly.

**Pull request merged**